transport.go

```diff
@@ -84,17 +84,16 @@ type Transport struct {
 	ptr *C.git_transport
 }
 
-// SmartProxyOptions gets a copy of the proxy options for this transport.
-func (t *Transport) SmartProxyOptions() (*ProxyOptions, error) {
+func (t *Transport) SmartRemoteConnectOptions() (*RemoteConnectOptions, error) {
```

Comment:
```suggestion
// SmartRemoteConnectOptions gets a copy of the proxy options for this transport.
func (t *Transport) SmartRemoteConnectOptions() (*RemoteConnectOptions, error) {
```

let's keep the comment so that https://pkg.go.dev/github.com/libgit2/git2go/v34 shows more information to developers.